gg/mark.go

```diff
@@ -59,6 +59,39 @@ func (m *markPath) mark(env *renderEnv, canvas *svg.SVG) {
 	drawPath(canvas, xs, ys, stroke, fill)
 }
 
+type markArea struct {
+	x, upper, lower, fill *scaledData
+}
+
+func rev(data []float64) []float64 {
```

Comment:
Consider spelling this out as "reversed", partly so that it's more obvious that this is not an in-place reversal (which I would spell "reverse").